### PR TITLE
add CBA_fnc_hashValues

### PR DESCRIPTION
**Pull request by Fusselwurm**
There is a `hashKeys` function  - but the corresponding function `hashValues` is missing

### CODE REVIEW

File: addons/hashes/fnc_hashValues.sqf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Author:
+    Fusselwurm
+---------------------------------------------------------------------------- */
+SCRIPT(hashValues);
+
+// -----------------------------------------------------------------------------
+params [["_hash", [[], []], [[]]]];
+
++(_hash select HASH_VALUES)
```
Comment (PabstMirror) [inline addons/hashes/fnc_hashValues.sqf:25]:
```suggestion
_hash select HASH_VALUES
```
`select` always creates a copy

**Review by PabstMirror — APPROVED** [review]

File: addons/hashes/fnc_hashValues.sqf (hunk 1 above)
Comment (mharis001) [reply addons/hashes/fnc_hashValues.sqf]: `ARRAY select NUMBER` returns a reference. For example:
```sqf
private _a = [[1], []]; 
private _b = _a select 0; 
_b pushBack 2; 
_a // -> [[1,2],[]]
```

Comment (Fusselwurm) [reply addons/hashes/fnc_hashValues.sqf:25]: ok, I removed the changes again :sweat_smile:  `+(x select y)` it is!

Comment (commy2) [reply addons/hashes/fnc_hashValues.sqf:25]: Consider a shallow copy:

``` 

[] + (_hash select HASH_VALUES

```
which will keep value array references as is what I would expect.

Comment (PabstMirror) [reply addons/hashes/fnc_hashValues.sqf:25]: oh no I'm an idiot; I was thinking of `select {}`

Comment (commy2) [inline addons/hashes/fnc_hashValues.sqf:25]:
```sqf
[] + (_hash select HASH_VALUES)
```

as described.

**Comment by commy2** [comment]
Now it does a deep copy.


```sqf
private _names = [] call CBA_fnc_hashCreate; 
[_names, "super", [1,2,3]] call CBA_fnc_hashSet; 
 
{ 
    _x pushBack 4; 
} forEach (_hash call CBA_fnc_hashValues); 
 
[_names, "super"] call CBA_fnc_hashGet // got [1,2,3], expected [1,2,3,4]
```

while:
```sqf
private _names = [] call CBA_fnc_hashCreate; 
[_names, "super", [1,2,3]] call CBA_fnc_hashSet; 
 
([_names, "super"] call CBA_fnc_hashGet) pushBack 4; 
 
[_names, "super"] call CBA_fnc_hashGet // got [1,2,3]
```

**Pull request merged**